spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/ScheduledTaskEndpoint.java

```diff
@@ -0,0 +1,179 @@
+	public List<ScheduledTaskInformation> invoke() {
+		return this.list;
+	}
+
+	@Override
+	public void configureTasks(ScheduledTaskRegistrar taskRegistrar) {
+		this.list.addAll(build(taskRegistrar.getFixedRateTaskList(), ScheduledType.FIXEDRATE));
+		this.list.addAll(build(taskRegistrar.getCronTaskList(), ScheduledType.CRON));
+		this.list.addAll(build(taskRegistrar.getFixedDelayTaskList(), ScheduledType.FIXEDDELAY));
+		this.list.addAll(build(taskRegistrar.getTriggerTaskList(), ScheduledType.TRIGGER));
+	}
+
+	private List<ScheduledTaskInformation> build(final List<? extends Task> tasks, final ScheduledType type) {
+		return tasks.stream().map(task -> {
+			ScheduledTaskInformation scheduledTaskInformation = new ScheduledTaskInformation();
+			scheduledTaskInformation.setType(type);
+			scheduledTaskInformation.setName(getGenericName(task.getRunnable()));
+			if (task instanceof IntervalTask) {
+				IntervalTask intervalTask = (IntervalTask) task;
+				scheduledTaskInformation.setInitialDelay(intervalTask.getInitialDelay());
+				scheduledTaskInformation.setInterval(intervalTask.getInterval());
+			}
+			if (task instanceof CronTask) {
+				CronTask cronTask = (CronTask) task;
+				scheduledTaskInformation.setExpression(cronTask.getExpression());
+			}
+			if (task instanceof TriggerTask) {
+				TriggerTask triggerTask = (TriggerTask) task;
+				scheduledTaskInformation.setTrigger(triggerTask.getTrigger().toString());
+			}
+			return scheduledTaskInformation;
+		}).collect(Collectors.toList());
+	}
+
+	private String getGenericName(Runnable runnable) {
+		if (runnable instanceof ScheduledMethodRunnable) {
+			return ((ScheduledMethodRunnable) runnable).getMethod().toGenericString();
+		}
+		return runnable.toString();
+	}
+
+	/**
+	 * Information for one scheduled task.
+	 */
+	@JsonInclude(Include.NON_EMPTY)
+	public class ScheduledTaskInformation {
+		private ScheduledType type;
+		private long interval;
+		private long initialDelay;
+		private String name;
+		private String expression;
+		private String trigger;
+
+		public ScheduledType getType() {
+			return this.type;
+		}
+
+		public void setType(ScheduledType type) {
+			this.type = type;
+		}
+
+		public long getInterval() {
+			return this.interval;
+		}
+
+		public void setInterval(long interval) {
+			this.interval = interval;
+		}
+
+		public long getInitialDelay() {
+			return this.initialDelay;
+		}
+
+		public void setInitialDelay(long initialDelay) {
+			this.initialDelay = initialDelay;
+		}
+
+		public void setName(String name) {
+			this.name = name;
+		}
+
+		public String getName() {
+			return this.name;
+		}
+
+		public void setExpression(String expression) {
+			this.expression = expression;
+		}
+
+		public String getExpression() {
+			return this.expression;
+		}
+
+		public void setTrigger(String trigger) {
+			this.trigger = trigger;
+		}
+
+		public String getTrigger() {
+			return this.trigger;
+		}
+	}
+
+	/**
+	 * Four different scheduled task type.
```

Comment:
This comment has details that is obvious and that can be changed in a future and will make this comment outdated. Also as a user who will read the API docs I don't need to know how many types here, because I can count them by myself.